packages/frontend/src/components/Explorer/ExploreTree/index.tsx

```diff
@@ -71,12 +72,30 @@ const ExploreTree: FC<ExploreTreeProps> = ({
     }, [explore, searchHasResults, isSearching]);
 
     const missingCustomMetrics = useMemo(() => {
-        const allTables = Object.keys(explore.tables);
-        return additionalMetrics.filter(
-            (metric) => !allTables.includes(metric.table),
-        );
+        return additionalMetrics.filter((metric) => {
+            const table = explore.tables[metric.table];
+            return (
+                !table ||
+                (metric.baseDimensionName &&
+                    !table.dimensions[metric.baseDimensionName])
```

Comment:
why did you change this ? I thought the problem was only on custom dimensions

Reply:
yeah, ended up noticing as well that if the base dimension for a custom metric was removed the custom metric wouldn't show as missing

https://github.com/lightdash/lightdash/assets/22939015/b396e478-3739-4077-aeb0-c6d6fe3b3436